src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -117,16 +139,39 @@ export class TelemetryReporter {
       }
       return error
     })
+    await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'errors'))
     await ensureFile(metaErrorFilePath)
-    await writeJson(metaErrorFilePath, { errors: errorsReport })
+    await writeJson(metaErrorFilePath, errorsReport)
   }
 
-  private async pendingFilesPaths() {
-    const pendingDataFiles = (await readdir(TelemetryReporter.PENDING_DATA_DIR)).filter(fileName => {
-      return fileName !== this.dataPendingLock.lockName
-    })
+  public async ensureTelemetryDirMazSize() {
+    const telemetryFiles = await readdirRecursive(TelemetryCollector.TELEMETRY_LOCAL_DIR)
```

Comment:
How about using `globby`? https://github.com/sindresorhus/globby

I think the following will do the trick:
```
import glob from 'globby'

public async ensureTelemetryDirMaxSize() {
  const telemetryFiles = await glob('**', cwd: TelemetryCollector.TELEMETRY_LOCAL_DIR)

  ...
}
```